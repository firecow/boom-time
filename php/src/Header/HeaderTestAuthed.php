<?
declare(strict_types=1);

namespace App\Header;

use App\Exceptions\DAOException;
use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Testing\RegressionTest;

class HeaderTestAuthed extends RegressionTest
{

    /**
     * @throws FileErrorException
     * @throws FileNotFoundException
     * @throws DAOException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $superGlobals->setHTTPAuthorization("Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJzaW1wbGVwaHAuY29tIiwiaWF0IjoxNTE5MzA0NTIyLCJleHAiOjE1MTkzOTA5MjIsInN1YiI6Ijc0NTA3NTFmMzA3YmM3NmQ5NjBmMmQ3OTJjNjZmNzIwIiwiaW1nIjoiaHR0cDovL2ZpcmVjb3cuZGsvaW1hZ2VzL2JnLnBuZyIsInVuYSI6IkZpcmVjb3cifQ.uG03s_iBmB2Zmd3ZBf-MAnlmyWPRSPsUT9FLoH7CSGk");
        $ctx = new ContextMock($superGlobals);
        $html = Header::getHeaderHTML($ctx);
        return $html;
    }
}
