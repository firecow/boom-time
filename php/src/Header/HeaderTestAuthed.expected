<img src="/photo/5b3e0bc586463d34800048a3">

<button onclick="return simplePHP.pushPageToHistory('/games/')">
    <i class="fas fa-gamepad"></i>
    <c>Games</c>
</button>
<button onclick="return simplePHP.pushPageToHistory('/myprofile/')">
    <i class="fas fa-user"></i>
    <c>Profile</c>
</button>
<button onclick="return simplePHP.pushPageToHistory('/error/')">
    <i class="fas fa-exclamation-circle"></i>
    <c>Error</c>
</button>
<button onclick="return simplePHP.clickLogout.run(event);">
    <i class="fas fa-sign-out-alt"></i>
    <c>Logout</c>
</button>